packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -153,7 +164,10 @@ function getLightSlottedContent(rootNodes: IrChildNode[], cxt: TransformerContex
                     // '' is the default slot name. Text nodes are always slotted into the default slot
                     const slotName =
                         node.type === 'Text' ? b.literal('') : bAttributeValue(node, 'slot');
+                    const { isSlotted } = cxt;
+                    cxt.isSlotted = ancestorIndices.length > 0 || node.type === 'Slot';
                     addLightDomSlotContent(slotName, [...ancestorIndices, i]);
+                    cxt.isSlotted = isSlotted;
```

Comment:
This is to prevent bookends from being omitted when not at the top-level. I'll add a comment.

Reply:
Makes sense. I kept trying to wrap my head around how some of the logic around VFragments would work at compile-time.

Reply:
Yes, it's also complex because the bookends only render when the fragment renders.

For example:

```
<template lwc:if={isFalse}>
  <span>yolo</span>
</template>
```

This will render without bookends. But if you used `isTrue`, it would render with bookends.

Reply:
Right! That